src/main/resources/avro/sequenceAnnotations.avdl

```diff
@@ -0,0 +1,137 @@
+@namespace("org.ga4gh.models")
+/**
+This protocol defines annotations on GA4GH genomic sequences It includes two
+types of annotations: continuous and discrete hierarchical.
+
+The discrete hierarchical annotations are derived from the Sequence Ontology
+(SO) and GFF3 work 
+
+   http://www.sequenceontology.org/gff3.shtml
+
+The goal is to be able to store annotations using the GFF3 and SO conceptual
+model, although there is not necessarly a one-to-one mapping in Avro records
+to GFF3 records.
+
+The minimum requirement is to be able to accurately represent the current
+state of the art annotation data and the full SO model.  Feature is the
+core generic record which corresponds to the a GFF3 record.
+*/
+protocol SequenceAnnotations {
+
+  import idl "common.avdl";
+  import idl "metadata.avdl";
+
+  /**
+  Type defining a collection of attributes associated with various protocol
+  records.  Each attribute is a name that maps to an array of one or more
+  values.  Values can be strings, external identifiers, or ontology terms.
+  Values should be split into the array elements instead of using a separator
+  syntax that needs to parsed.
+  */
+  // TODO: how are multiple instances of a given attribute vs multiple values
+  // for an attribute distinguished
+  record Attributes {
+    map<array<union {string, ExternalIdentifier, OntologyTerm}>> vals = {};
+  }
+  
+  /**
+  Node in the annotation graph that annotates a contiguous region of a
+  sequence.
+  */
+  record Feature {
+    /**
+    Id of this annotation node.
+    */
+    string id;
+
+    /**
+    Parent Id of this node. Set to empty string if node has no parent.
+    */
+    string parentId;
+
+    /**
+    Ordered array of Child Ids of this node.
+    Since not all child nodes are ordered by genomic coordinates,
+    this can't always be reconstructed from parentId's of the children alone.
+    */
+
+    array<string> childIds = [];
+
+    /**
+    Identifier for the containing feature set.
+    */
+    string featureSetId;
+
+    /**
+    The reference on which this feature occurs.
+    (e.g. `chr20` or `X`)
+    */
+    string referenceName;
```

Comment:
Issue is:
https://github.com/ga4gh/schemas/issues/593

Maciek Smuga-Otto <notifications@github.com> writes:
> In src/main/resources/avro/sequenceAnnotations.avdl:
> 
> > +    Since not all child nodes are ordered by genomic coordinates,
> > +    this can't always be reconstructed from parentId's of the children alone.
> > +    */
> > +
> > +    array<string> childIds = [];
> > +
> > +    /**
> > +    Identifier for the containing feature set.
> > +    */
> > +    string featureSetId;
> > +
> > +    /**
> > +    The reference on which this feature occurs.
> > +    (e.g. `chr20` or `X`)
> > +    */
> > +    string referenceName;
> 
> I now understand your point, @reece - thanks for elaborating! And yes, the
> referenceId should exclusively be used. I'd like to propose this (using
> reference ID as opposed to name) uniformly across all API endpoints as a
> separate PR, though, as it's a cross-cutting issue.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub*
>